diff -uNr simsam-0.1.8.ori/lib/instrument.cpp simsam-0.1.8/lib/instrument.cpp
--- simsam-0.1.8.ori/lib/instrument.cpp	2009-05-17 22:27:08.000000000 +0200
+++ simsam-0.1.8/lib/instrument.cpp	2009-05-17 22:58:34.000000000 +0200
@@ -6,6 +6,8 @@
 #include <list>
 #include <string>
 
+#include <algorithm>
+
 Instrument Instrument::instruments[16];
 
 int Instrument::currentId = 0;
diff -uNr simsam-0.1.8.ori/lib/output.cpp simsam-0.1.8/lib/output.cpp
--- simsam-0.1.8.ori/lib/output.cpp	2009-05-17 22:27:08.000000000 +0200
+++ simsam-0.1.8/lib/output.cpp	2009-05-17 22:57:41.000000000 +0200
@@ -18,6 +18,8 @@
 #include <qdom.h>
 #include <qfile.h>
 
+#include <algorithm>
+
 /* generic stuff */
 
 float Output::masterVolume = 1.0f;
